src/postgres/tree/PostgresFunctionsTreeItem.ts

```diff
@@ -47,29 +105,39 @@ export class PostgresFunctionsTreeItem extends AzureParentTreeItem<ISubscription
             left join pg_language l on p.prolang = l.oid
             where n.nspname not in ('pg_catalog', 'information_schema')
                 ${this.parent.parent.supportsStoredProcedures() ? "and p.prokind = 'f'" : '' /* Only select functions, not stored procedures */}
+                ${functionName ? `and p.proname = '${functionName}'` : '' /* Only select functions that match the given name (if provided) */}
             order by name;`;
 
         const queryResult: QueryResult = await client.query(functionsQuery);
-        const rows: IPostgresProceduresQueryRow[] = queryResult.rows || [];
+        return queryResult.rows || [];
+    }
 
-        const allNames: Set<string> = new Set();
-        const duplicateNames: Set<string> = new Set();
-        for (const row of rows) {
-            if (allNames.has(row.name)) {
-                duplicateNames.add(row.name);
-            } else {
-                allNames.add(row.name);
-            }
-        }
+    private async listSchemas(): Promise<string[]> {
+        const client = new Client(this.clientConfig);
+        await client.connect();
 
-        return rows.map(row => new PostgresFunctionTreeItem(
-            this,
-            row,
-            duplicateNames.has(row.name)
-        ));
+        const schemaQuery: string = `select n.nspname as schema
+            from pg_namespace as n
+            where n.nspname not in ('pg_catalog', 'information_schema', 'pg_toast', 'pg_temp_1', 'pg_toast_temp_1')`;
+
+        const queryResult: QueryResult = await client.query(schemaQuery);
+        return queryResult.rows ? queryResult.rows.map(row => row.schema) : [];
     }
 
-    public isAncestorOfImpl(contextValue: string): boolean {
-        return contextValue === PostgresFunctionTreeItem.contextValue;
+    private validateFunctionName(name: string, existingFunctions: Set<string>): string | undefined {
+        if (existingFunctions.has(name)) {
+            return localize('functionAlreadyExists', 'A function named "{0}" already exists.', name);
```

Comment:
I would write a function named "{0}" with schema "{1}" already exists since it might be confusing as to why you can have duplicates sometimes but not others.